### PR TITLE
Fix/kotlin to java translation

### PR DESCRIPTION
**Pull request by bartekryba**
Fixes for kotlin -> java transpilation.

### CODE REVIEW

**Comment by bartekryba** [comment]
Other problems I found:

1. We could change the display name of the library.
Zrzut ekranu 2022-08-11 o 15 30 53

2. Dokka generates these getters for NetUrls, although they're not accessible from Java.

Zrzut ekranu 2022-08-11 o 15 27 14

3. Many classes remain undocumented, as well as automatically generated classes, annotated with `@file:JvmName` 

4. There are lots of classes in `com.swmansion.starknet.data.types`, we might consider making the project structure more Jvm-like, by not keeping so much in one package/namespace

**Comment by Solpatium** [comment]
>We could change the display name of the library.

I agree, let's do it.

>Dokka generates these getters for NetUrls, although they're not accessible from Java.
What if we use JvmStatic decorator?

>Many classes remain undocumented, as well as automatically generated classes, annotated with @file:JvmName
We can focus on documentation post release

>There are lots of classes in com.swmansion.starknet.data.types

I think we can divide them into some subpackages.

**Comment by bartekryba** [comment]
> What if we use JvmStatic decorator?
It's not possible together with `const` keyword

**Comment by bartekryba** [comment]
> I think we can divide them into some subpackages.

In this pr for initial release, or later?

**Review by cptartur — COMMENTED** [review]
Just a single comment. Post release we should consider doing a major changes to the encapsulation of the modules in our library - right now pretty much everything is public and it pollutes both docs and namespace. 

Also I agree we're in dire need of docs improvement - there are multiple public classes that are left undocumented and that's not good.

File: lib/build.gradle.kts [hunk 1]
```diff
@@ -39,6 +39,8 @@ tasks.withType<DokkaTask>().configureEach {
             includes.from("lib.md")
         }
     }
+
+    moduleName.set("starknet-jvm")
```
Comment (cptartur) [inline lib/build.gradle.kts:43]: Setting this parameter causes for the `lib.md` to not be included as part of docs for some reason.

File: lib/src/main/kotlin/com/swmansion/starknet/data/types/transactions/Transaction.kt [hunk 2]
```diff
@@ -131,25 +131,27 @@ data class DeclareTransaction(
...
-        contractAddress,
-        entryPointSelector,
-        StarknetCurve.pedersenOnElements(calldata),
-        maxFee,
-        chainId.value,
-    )
-
-    return InvokeTransaction(contractAddress, calldata, entryPointSelector, hash, maxFee, version, signature, nonce)
+object TransactionFactory {
+    fun makeInvokeTransaction(
```
[the remark this answers is not in the document]
Comment (cptartur) [reply lib/src/main/kotlin/com/swmansion/starknet/data/types/transactions/Transaction.kt:135]: This will be addressed in separate PR

**Pull request merged**